cluster-template-patches/calico-static-ip-and-mac.yaml

```diff
@@ -0,0 +1,16 @@
+resources:
+  - cluster-template.yaml
+patches:
+  - target:
+      kind: VirtinkMachineTemplate
+    patch: |-
+      apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+      kind: VirtinkMachineTemplate
+      metadata:
+        name: "not-used"
+      spec:
+        template:
+          metadata:
+            annotations:
+              cni.projectcalico.org/ipAddrs: '["$IP_ADDRESS"]'
```

Comment:
Why not add them to templates rather than merging them on the fly?

Reply:
capch provides different template for each CNI, or user add it to template?

Reply:
We can use templates from capch accordingly by CNI flag?

Reply:
capch didn't provide templates for each CNI, in capch user have to download the common template and add annotations manually.  knest did this automatically for Calico and Kube-ovn now.

Reply:
Then maybe we should add these templates in capch, and use them accordingly in knest?

Reply:
this will make capch have many templates().  at least 4 templates need to be added. they are `cluster-template-cdi.yaml` + calico/kube-ovn，`cluster-template-cdi-internal.yaml` + calico/kube-ovn. The design is if user use capch directly, they have to add annotations by themselves, if user use knest, knest did it.

Reply:
Yes, there will be many templates, but I think it's better we add annotations for CNIs we support than let user add them since it's not a trivial task.